src/components/notifcations/Notifications.tsx

```diff
@@ -0,0 +1,43 @@
+import { UserModel } from "../../lib/schemas/user.schema";
+
+export async function addNotification(userId: string, newNotification: string) {
+    const user = await UserModel.findById(userId);
+    if (user) {
+      if (user.notifications.length >= 3) {
+        user.notifications.shift(); // Remove the oldest notification
+      }
+      user.notifications.push(newNotification); // Add the new notification
+      await user.save();
+    }
+  }
```

Comment:
This function is an API related one, but being used in the same file as a React component, which I think can be really confusing. We should move this to an appropriate place in the API folder?